models/training-tuning-scripts/root-cause-models/sequence_classifier.py

```diff
@@ -0,0 +1,254 @@
+
+import cupy
+import torch
+from dataloader import DataLoader
+from dataset import Dataset
+from torch.utils.dlpack import to_dlpack
+from tqdm import trange
+from transformers import AdamW
+
+import cudf
+from cudf.core.subword_tokenizer import SubwordTokenizer
+
+log = logging.getLogger(__name__)
+
+
+class SequenceClassifier(ABC):
+    """
+    Sequence Classifier using BERT. This class provides methods for training/loading BERT models, evaluation and
+    prediction.
+    """
+
+    def __init__(self):
+        self._device = None
+        self._model = None
+        self._optimizer = None
+        self._hashpath = self._get_hash_table_path()
+
+    @abstractmethod
+    def predict(self, input_data, max_seq_len=128, batch_size=32, threshold=0.5):
+        pass
+
+    def train_model(
+        self,
+        train_data,
+        labels,
+        learning_rate=3e-5,
+        max_seq_len=128,
+        batch_size=32,
+        epochs=5,
+    ):
+        """
+        Train the classifier
+
+        :param train_data: text data for training
+        :type train_data: cudf.Series
+        :param labels: labels for each element in train_data
+        :type labels: cudf.Series
+        :param learning_rate: learning rate
+        :type learning_rate: float
+        :param max_seq_len: Limits the length of the sequence returned by tokenizer. If tokenized sentence is shorter
+            than max_seq_len, output will be padded with 0s. If the tokenized sentence is longer than max_seq_len it
+            will be truncated to max_seq_len.
+        :type max_seq_len: int
+        :param batch_size: batch size
+        :type batch_size: int
+        :param epoch: epoch, default is 5
+        :type epoch: int
+
+        Examples
+        --------
+        >>> from cuml.preprocessing.model_selection import train_test_split
+        >>> emails_train, emails_test, labels_train, labels_test = train_test_split(train_emails_df,
+                                                                                    'label',
+                                                                                    train_size=0.8)
+        >>> sc.train_model(emails_train, labels_train)
+        """
+        train_gdf = cudf.DataFrame()
+        train_gdf["text"] = train_data
+        train_gdf["label"] = labels
+
+        train_dataset = Dataset(train_gdf)
+        train_dataloader = DataLoader(train_dataset, batchsize=batch_size)
+
+        self._config_optimizer(learning_rate)
+        self._model.train()  # Enable training mode
+        self._tokenizer = SubwordTokenizer(self._hashpath, do_lower_case=True)
+
+        for _ in trange(epochs, desc="Epoch"):
+            tr_loss = 0  # Tracking variables
+            nb_tr_examples, nb_tr_steps = 0, 0
+            for df in train_dataloader.get_chunks():
+                b_input_ids, b_input_mask = self._bert_uncased_tokenize(df["text"], max_seq_len)
+
+                b_labels = torch.tensor(df["label"].to_array())
```

Comment:
`to_array()` is no longer in cudf 22.08. should `to_numpy()`

Reply:
updated for later versions of cudf